### PR TITLE
Add `maxSplit` parameter to Java binding for `strings:split`

### PR DESCRIPTION
**Pull request by ttnghia**
Currently, Java binding for `strings::split` calls the API with the default max split value. This PR adds a parameter for it, allowing to specify max split from Java.

This is non-breaking because the Java code implicitly uses the default max split value.

### CODE REVIEW

**Comment by ttnghia** [comment]
Rerun tests.

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -2331,13 +2331,27 @@ public final ColumnVector stringLocate(Scalar substring, int start, int end) {
    * Null string entries return corresponding null output columns.
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:2331]: 2022 copyrights

Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:2331]: Oh yeah.

**Review by jlowe — APPROVED** [review]

**Comment by ttnghia** [comment]
@gpucibot merge

**Pull request merged**